opal/mca/hwloc/base/hwloc_base_util.c

```diff
@@ -1018,7 +1018,7 @@ static char *bitmap2rangestr(int bitmap)
     size_t i;
     int range_start, range_end;
     bool first, isset;
-    char tmp[BUFSIZ];
+    char tmp[BUFSIZ - 1];
```

Comment:
What warning does this fix? It seems weird to fix a warning by reducing a buffer size...

Reply:
It was supposed to fix these two:
```
hwloc_base_util.c:1050: warning: ‘strncat’ output may be truncated
copying between 0 and 8191 bytes from a string
of length 8191 [-Wstringop-truncation]
base/hwloc_base_util.c:1077: warning: ‘strncat’ output may be truncated
copying between 0 and 8191 bytes from a string
of length 8191 [-Wstringop-truncation]
```
From the strcat man page:
```
If the length of the C string in source is less than num, only the content up to the terminating null-character is copied.
```

I think it's just the compiler being extra verbose with strcat. I changed it to `snprintf()` to leave the buffer alone. Thanks